### PR TITLE
async.race() method

### PR DESCRIPTION
**Pull request by jorgebay**
Based on the discussion from #568 .
Contains mocha-based tests and readme update.

### CODE REVIEW

File: lib/async.js [hunk 1]
```diff
@@ -1247,6 +1247,27 @@
         });
     };
 
+    async.race = function (tasks, callback) {
+        callback = _once(callback || noop);
+        if (!_isArray(tasks)) {
+            return callback(new TypeError('First argument to waterfall must be an array of functions'));
```
Comment (megawac) [inline lib/async.js]: typo, change waterfall to race

File: lib/async.js [hunk 2]
```diff
@@ -1247,6 +1247,27 @@
...
+            return callback(null);
+        }
+        for (var i = 0; i < tasks.length; i++) {
+            tasks[i](only_once(done));
+        }
+        var called = false;
+        function done(err, result) {
+            if (called) {
+                return;
+            }
```
Comment (megawac) [inline lib/async.js]: no need for `called` or this condition. once will handle it

**Comment by megawac** [comment]
:+1:

**Comment by aearly** [comment]
@megawac Should we merge this now, or wait until the modularization is complete?  (That branch is almost ready to be merged, just some package questions remaining).

**Comment by megawac** [comment]
Either way is fine, I can rebase/cherry-pick any commits onto the
modularizations branch

**Comment by jorgebay** [comment]
@aearly @megawac I could send a new pull request against the modularization branch and close this one, what do you think?

**Comment by aearly** [comment]
That would be really helpful.

**Comment by aearly** [comment]
Modularization is now on master, so no weird branch to consider now!

**Comment by jorgebay** [comment]
Thanks for the heads up, I will submit a new pr soon.

**Comment by megawac** [comment]
Cool @jorgebay. By the way, it might make sense to implement this via `async.some` as you can accomplish the same result with `some`

**Comment by aearly** [comment]
Yeah, but heads up, we also want to change the argument for `some` and family. #118

**Pull request closed without merging**